parser.ts

```diff
@@ -321,6 +330,11 @@ export class Parser {
     return tracer.spanSync('lineNumber', () => {
       const prevLineNo = this.lineNo;
       if (this.match(TokenKind.DecimalLiteral)) {
+        this.lineSource = new LineSource(
```

Comment:
It's tempting to track this in a separate property. But `rowEnding` needs to return the same thing regardless of whether we're parsing a `Line` or a `Command`, which implies that we should replace `this.line` completely.